components/partials/spot/trading/trade.vue

```diff
@@ -110,20 +60,114 @@
+            :label="$t('trade.amount')"
+            :custom-handler="true"
+            :max-decimals="market ? market.quantityDecimals : 6"
+            :placeholder="amountStep"
+            type="number"
+            :step="amountStep"
+            min="0"
+            data-cy="trading-page-amount-input"
+            show-addon
+            @input="onAmountChange"
+            @blur="onAmountBlur"
+          >
+            <span slot="addon">{{
+              market.baseToken.symbol.toUpperCase()
+            }}</span>
+            <div
+              slot="context"
+              class="text-xs text-gray-400 flex items-center font-mono"
+            ></div>
+          </v-input>
+          <v-input
+            ref="input-amount"
+            :key="`quoteAmount-${quoteAmountKey}`"
+            v-model="form.quoteAmount"
+            :custom-handler="true"
+            :max-decimals="market ? market.quantityDecimals : 6"
+            :placeholder="amountStep"
+            type="number"
+            :step="amountStep"
+            min="0"
+            data-cy="trading-page-amount-input"
+            show-prefix
+            show-addon
+            @input="onQuoteAmountChange"
+            @blur="onQuoteAmountBlur"
+            @input-max="() => onProportionalQuantitySelected(100)"
+          >
+            <span slot="prefix">≈</span>
+            <span slot="addon">{{
+              market.quoteToken.symbol.toUpperCase()
+            }}</span>
+            <div
+              slot="context"
+              class="text-xs text-gray-400 flex items-center font-mono"
+            >
+              <span
+                class="mr-1 cursor-pointer"
+                @click.stop="onProportionalQuantitySelected(25)"
+              >
+                25%
+              </span>
+              <span
+                class="mr-1 cursor-pointer"
+                @click.stop="onProportionalQuantitySelected(50)"
+              >
+                50%
+              </span>
+              <span
+                class="mr-1 cursor-pointer"
+                @click.stop="onProportionalQuantitySelected(75)"
+              >
+                75%
+              </span>
+              <span
+                class="cursor-pointer"
+                @click.stop="onProportionalQuantitySelected(100)"
+              >
+                100%
+              </span>
+            </div>
+          </v-input>
+        </div>
+        <span
+          v-if="amountError"
+          class="text-2xs font-semibold text-red-500"
+          data-cy="trading-page-amount-error-text-content"
+        >
+          {{ amountError }}
+        </span>
         <span
           v-if="priceError"
-          class="text-red-500 font-semibold text-2xs"
           data-cy="trading-page-price-error-text-content"
+          class="text-2xs font-semibold text-red-500"
         >
           {{ priceError }}
         </span>
       </div>
     </div>
+    <div>
+      <AdvancedSettings
```

Comment:
Not a comment on your PR perse but I've noticed we now have this slippage tolerance input field that is both being used here and in swaps, might be a good idea to re-use between the two. https://github.com/InjectiveLabs/injective-dex/blob/dev/components/partials/swap/advanced-settings.vue

Reply:
I actually used the swap `advanced-settings.vue` originally as a template since it looks dope and then customized it a bit to match the trade order form. then, I got a new design from Ruth that basically reduced the slippage component down to an input field since the real estate is pretty tight. not sure if she designed the slippage tolerance for swap. could be worth asking her whether she wants swap to match her new design for trade order form. I'm actually ok with them being different since trade order form has less real estate, which I'm thinking was her main consideration, and not a limitation for swap

Reply:
I discussed with terence, for now we'll leave them separate and once both of our changes have been deployed/finalized we can look into maybe cleaning up a bit by reusing some logic via a shared component maybe. Something along those lines. :)